ecmascript/transforms/compat/src/es2017/async_to_generator.rs

```diff
@@ -362,6 +362,37 @@ impl Fold for Actual {
         }
     }
 
+    fn fold_module_item(&mut self, item: ModuleItem) -> ModuleItem {
+        let item = match item {
+            // if fn is ExportDefaultDecl, fn is not FnDecl but FnExpr
+            ModuleItem::ModuleDecl(ModuleDecl::ExportDefaultDecl(export_default)) => {
+                if let DefaultDecl::Fn(expr) = export_default.decl {
+                    let expr = if !expr.function.is_async {
+                        expr
+                    } else {
+                        let function = self.fold_fn(expr.ident.clone(), expr.function, true);
+                        FnExpr {
+                            ident: expr.ident,
+                            function,
+                        }
+                    };
+
+                    return ModuleItem::ModuleDecl(ModuleDecl::ExportDefaultDecl(
+                        ExportDefaultDecl {
+                            decl: expr.into(),
+                            ..export_default
+                        },
+                    ));
+                } else {
+                    return ModuleItem::ModuleDecl(ModuleDecl::ExportDefaultDecl(export_default));
```

Comment:
I think `return` should be removed as `async`s in `export default class Foo {}` will not be folded otherwise.

Reply:
Uh, makes sense. Removing this makes match arm return type differences - updated code to fold for the remaining cases other than defaulyt expr.